test/units/staking/stakeManager/StakeManager.Staking.js

```diff
@@ -0,0 +1,786 @@
+        const amounts = walletAmounts[wallets[3].getAddressString()]
+        testStake(
+          wallets[3].getChecksumAddressString(),
+          wallets[3].getPublicKeyString(),
+          amounts.amount,
+          amounts.stakeAmount,
+          1
+        )
+      })
+
+      describe('when other user stakes beyond validator threshold', function() {
+        testStakeRevert(
+          wallets[4].getChecksumAddressString(),
+          wallets[4].getPublicKeyString(),
+          web3.utils.toWei('100'),
+          web3.utils.toWei('100'),
+          true
+        )
+      })
+    })
+
+    describe('consecutive stakes', function() {
+      before(freshDeploy)
+
+      it('validatorId must increment 1 by 1', async function() {
+        const _wallets = [wallets[1], wallets[2], wallets[3]]
+        let expectedValidatorId = 1
+        for (const wallet of _wallets) {
+          await doStake(wallet, { approveAmount: web3.utils.toWei('100'), stakeAmount: web3.utils.toWei('100') }).call(this)
+
+          const validatorId = await this.stakeManager.getValidatorId(wallet.getAddressString())
+          assertBigNumberEquality(expectedValidatorId, validatorId)
+          expectedValidatorId++
+        }
+      })
+    })
+
+    describe('stake with heimdall fee', function() {
+      before(freshDeploy)
+
+      testStake(
+        wallets[0].getChecksumAddressString(),
+        wallets[0].getPublicKeyString(),
+        web3.utils.toWei('200'),
+        web3.utils.toWei('150'),
+        1,
+        web3.utils.toWei('50')
+      )
+    })
+  })
+
+  describe('unstake', function() {
+    describe('when user unstakes right after stake', async function() {
+      const user = wallets[2].getChecksumAddressString()
+      const amounts = walletAmounts[wallets[2].getAddressString()]
+
+      before('Fresh deploy', freshDeploy)
+      before('Validator threshold and dynasty', async function() {
+        await this.stakeManager.updateValidatorThreshold(3, {
+          from: owner
+        })
+
+        await this.stakeManager.updateDynastyValue(2, {
+          from: owner
+        })
+      })
+
+      before(doStake(wallets[2]))
+      before(async function() {
+        this.validatorId = await this.stakeManager.getValidatorId(user)
+
+        const validator = await this.stakeManager.validators(this.validatorId)
+        this.reward = validator.reward
+        this.afterStakeBalance = await this.stakeToken.balanceOf(user)
+      })
+
+      it('must unstake', async function() {
+        this.receipt = await this.stakeManager.unstake(this.validatorId, {
+          from: user
+        })
+      })
+
+      it('must emit UnstakeInit', async function() {
+        await expectEvent.inTransaction(this.receipt.tx, StakingInfo, 'UnstakeInit', {
+          amount: amounts.stakeAmount,
+          validatorId: this.validatorId,
+          user
+        })
+      })
+
+      // for some reason DummyERC20 doesn't have Transfer event
```

Comment:
if you are using truffle log parser then give a try to use test token instead od using DummyErc20. Due to the same event `Transfer` in both erc721/erc20, I wasn't able to parse that log so removed in Dummy one.

Reply:
I see, i'll get it back then